apps/src/aichat/redux/aichatRedux.ts

```diff
@@ -0,0 +1,55 @@
+import {createSlice, PayloadAction} from '@reduxjs/toolkit';
+import {ChatCompletionMessage} from '../types';
+import {initialChatMessages} from '../constants';
+
+const registerReducers = require('@cdo/apps/redux').registerReducers;
+
+export interface AichatState {
+  // All user and assistant chat messages - includes too personal and inappropriate user messages.
+  // Messages will be logged and stored.
+  chatMessages: ChatCompletionMessage[];
+  // New user message that is submitted via the user chat message editor.
+  newUserMessage: string;
+  // Denotes whether we are waiting for a chat completion response from the backend
+  isWaitingForChatResponse: boolean;
+  // Denotes whether we should show the warning modal
+  showWarningModal: boolean;
+}
+
+const initialState: AichatState = {
+  newUserMessage: '',
+  chatMessages: initialChatMessages,
+  isWaitingForChatResponse: false,
```

Comment:
Could be a separate change - but I think we probably also want to block the user from submitting multiple messages in a row while waiting from a response? Might need to check with product/curriculum, but if so, we can probably use the isWaitingForChatResponse state to prevent the user from submitting another message.

Reply:
Great idea! I'll implement in a follow-up. Thanks!